source/MaterialXRender/Util.cpp

```diff
@@ -35,6 +35,42 @@ ShaderPtr createConstantShader(GenContext& context,
     return createShader(shaderName, context, output);
 }
 
+ShaderPtr createGammaShader(GenContext& context,
+                            DocumentPtr stdLib,
+                            const string& shaderName,
+                            const Color3& gamma,
+                            bool verticalFlip)
+{
+    // Construct the gamma correct nodegraph. Set to have a vertical flip.
+    DocumentPtr doc = createDocument();
+    doc->importLibrary(stdLib);
+
+    NodeGraphPtr nodeGraph = doc->addNodeGraph();
+    NodePtr texcoordNode = nodeGraph->addNode("texcoord", "texcoord", "vector2");
+    NodePtr multiplyNode = nodeGraph->addNode("multiply", "multiply", "vector2");
+    multiplyNode->setNodeDefString("ND_multiply_vector2");
+    multiplyNode->setConnectedNode("in1", texcoordNode);
+    NodePtr addNode = nodeGraph->addNode("add", "add", "vector2");
+    addNode->setNodeDefString("ND_add_vector2");
+    addNode->setConnectedNode("in1", multiplyNode);
+    NodePtr imageNode = nodeGraph->addNode("image", "image", "color3");
+    imageNode->setConnectedNode("texcoord", addNode);
+    NodePtr rangeNode = nodeGraph->addNode("range", EMPTY_STRING, "color3");
+    rangeNode->setConnectedNode("in", imageNode);
+    rangeNode->setInputValue("gamma", gamma);
+    OutputPtr output = nodeGraph->addOutput();
+    output->setConnectedNode(rangeNode);
+
+    if (verticalFlip)
```

Comment:
Why is the verticalFlip needed?

Reply:
We read the frame buffer from top to bottom so we need a flip (for OpenGL convention). If we read from bottom to top then no flip or if we draw with non-OpenGL then it could be bottom to top.